### DISCUSSION TITLE
Installing on Debian 12

### DISCUSSION THREAD

**Discussion by Corn-101** [inventree/InvenTree#8666] [Q&A]
Hi,

So I installed newest Debian V12 and later noticed that only v11 is supported so had some troubles in installing Inventree. The single line thing didn't work, but was able to install by the manual steps when first manually installing libssl1.1 as this was replaced in v12 with libssl3. "apt-get install inventree" went through without errors but I noticed that it gave warnings that config.yaml was not found and maybe due that the inventree didn't work. I'm not sure is the config.yaml only problem, but should the installation script have created that file? I had Apache2 installed as well, so does it conflict with the nginx if they both listen port 80? 

Thank you in advance

**Comment by Corn-101** [comment]
I killed the apache2 and I'm bit more closer now. However when running command:
`inventree run invoke version`
I get following results, which doesn't seem right. I installed manually django by apt install python3-django, but still same result. What might be the issue?

`
Traceback (most recent call last):
  File "/opt/inventree/env/bin/invoke", line 8, in <module>
    sys.exit(program.run())
             ^^^^^^^^^^^^^
  File "/opt/inventree/env/lib/python3.11/site-packages/invoke/program.py", line 398, in run
    self.execute()
  File "/opt/inventree/env/lib/python3.11/site-packages/invoke/program.py", line 583, in execute
    executor.execute(*self.tasks)
  File "/opt/inventree/env/lib/python3.11/site-packages/invoke/executor.py", line 140, in execute
    result = call.task(*args, **call.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/inventree/env/lib/python3.11/site-packages/invoke/tasks.py", line 138, in __call__
    result = self.body(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/inventree/tasks.py", line 1090, in version
    import src.backend.InvenTree.InvenTree.version as InvenTreeVersion
  File "/opt/inventree/src/backend/InvenTree/InvenTree/version.py", line 15, in <module>
    import django
ModuleNotFoundError: No module named 'django'
`

**Reply by SchrodingersGat** [reply]
https://docs.inventree.org/en/stable/faq/#no-module-named-django

Likely the virtual environment has not been configured properly as part of the install process, if you are getting this message? 

>  but was able to install by the manual steps 

What steps precisely did you follow?

**Reply by Corn-101** [reply]
Yeah, I guess that's the case. Should the installer script do that virtual environment configuration as well or should it be done manually? Should it also configure the database and config.yaml?

What I meant by the manual install is the steps below the installer script:
https://docs.inventree.org/en/latest/start/installer/#manual-install
Noticed that it's for Ubuntu so I did following to uninstall the Inventree:
`dpkg -r inventree
dpkg -P inventree`
And then used the installer script. As I had already manually installed dependencies and copied the config.yaml to the folder from the source, installation went without errors. Still that `inventree run invoke version`is giving that error. Also the installation script didn't alter the config.yaml at all. So I'm really stumbled that what am I missing. So what are the steps to be done before or after the installation script?: `wget -qO install.sh https://get.inventree.org && bash install.sh`

**Reply by Corn-101** [reply]
Well, I took the Docker image and got it running. However when I changed the `INVENTREE_SITE_URL`to `INVENTREE_SITE_URL=192.168.1.102` it doesn't accept any credentials anymore. No error message whatsoever and logs neither show any issues.
`inventree-server    | 192.168.1.102 - - [20/Dec/2024:16:00:58 +0000] "GET /accounts/login/?next=/ HTTP/1.1" 200 8637 "http://192.168.1.102/accounts/login/" "Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0"`

When I intentionally input wrong username the logon screen gives error of wrong username or password. Any idea what might be wrong?

**Reply by Corn-101** [reply]
decided to pull the latest .17 version and now it's forcing https so can't even get to the logon screen...
edit: when I change the this back to `INVENTREE_SITE_URL="http://inventree.localhost"` from .env it works. Why is it that `INVENTREE_SITE_URL="http://192.168.1.102"` doesn't work? I need the access from other computer from my internal network as well.
edit2: Changed the site_url from Caddyfile as well. Now I can access from other computer on network, but it wont let me in from the log screen. None of the credentials work. I'm stumbled. 

edit3: I decided to start from fresh setup to overrule any issues maybe caused by previous actions. So completely new Debian 12 installation, Inventree latest docker, same thing. Can log default inventree.localhost, but if I change that to `INVENTREE_SITE_URL="http://192.168.1.102"` I can access to logon page, but with correct credentials it will not open the index page. Not even if I manually add to the address http://192.168.1.102/index/
What am I missing?
